api/channel.go

```diff
@@ -791,126 +648,73 @@ func removeMember(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	sc := app.Srv.Store.Channel().Get(channelId, true)
-	cmc := app.Srv.Store.Channel().GetMember(channelId, c.Session.UserId)
-	ouc := app.Srv.Store.User().Get(userIdToRemove)
-
-	if oresult := <-ouc; oresult.Err != nil {
-		c.Err = model.NewLocAppError("removeMember", "api.channel.remove_member.user.app_error", nil, "")
+	var channel *model.Channel
+	var err *model.AppError
+	if channel, err = app.GetChannel(channelId); err != nil {
+		c.Err = err
 		return
-	} else {
-		oUser := oresult.Data.(*model.User)
-
-		if cresult := <-sc; cresult.Err != nil {
-			c.Err = cresult.Err
-			return
-		} else if cmcresult := <-cmc; cmcresult.Err != nil {
-			c.Err = cmcresult.Err
-			return
-		} else {
-			channel := cresult.Data.(*model.Channel)
-
-			if channel.Type == model.CHANNEL_OPEN && !HasPermissionToChannelContext(c, channel.Id, model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS) {
-				return
-			}
-
-			if channel.Type == model.CHANNEL_PRIVATE && !HasPermissionToChannelContext(c, channel.Id, model.PERMISSION_MANAGE_PRIVATE_CHANNEL_MEMBERS) {
-				return
-			}
-
-			if err := RemoveUserFromChannel(userIdToRemove, c.Session.UserId, channel); err != nil {
-				c.Err = model.NewLocAppError("updateChannel", "api.channel.remove_member.unable.app_error", nil, err.Message)
-				return
-			}
-
-			c.LogAudit("name=" + channel.Name + " user_id=" + userIdToRemove)
-
-			go app.PostUserAddRemoveMessage(c.Session.UserId, channel.Id, channel.TeamId, fmt.Sprintf(utils.T("api.channel.remove_member.removed"), oUser.Username), model.POST_ADD_REMOVE)
+	}
 
-			result := make(map[string]string)
-			result["channel_id"] = channel.Id
-			result["removed_user_id"] = userIdToRemove
-			w.Write([]byte(model.MapToJson(result)))
-		}
+	if channel.Type == model.CHANNEL_OPEN && !HasPermissionToChannelContext(c, channel.Id, model.PERMISSION_MANAGE_PUBLIC_CHANNEL_MEMBERS) {
```

Comment:
Wondering if we should move the validation of this kind of permissions to the app layer